waku/v2/node/waku_node.nim

```diff
@@ -601,11 +601,28 @@ proc mountLightPush*(node: WakuNode) {.async.} =
   var pushHandler: PushMessageHandler
   if node.wakuRelay.isNil():
     debug "mounting lightpush without relay (nil)"
-    pushHandler = proc(peer: PeerId, pubsubTopic: string, message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
+    pushHandler = proc(peer: PeerId, pubsubTopic: Option[string], message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
       return err("no waku relay found")
   else:
-    pushHandler = proc(peer: PeerId, pubsubTopic: string, message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
-      discard await node.wakuRelay.publish(pubsubTopic, message.encode().buffer)
+    pushHandler = proc(peer: PeerId, pubsubTopic: Option[string], message: WakuMessage): Future[WakuLightPushResult[void]] {.async.} =
+      let topic =
```

Comment:
Should this code be extracted somewhere into a proc? It seems exact same code repeats in `waku_node.nim` and `waku_filter/client.nim` - not a fan of premature optimization, but this seems like a generally useful piece of code 

`pubsubTopicOrShard(pubsubTopic: Option[string], contentTopic: string): string` or something like this?

Reply:
Good point. It's already used in 5 places.